frontend/app/routes/protected/person-case/birth-details.tsx

```diff
@@ -0,0 +1,275 @@
+                v.string(t('protected:birth-details.city.required-city')),
+                v.trim(),
+                v.nonEmpty(t('protected:birth-details.city.required-city')),
+                v.maxLength(maxStringLength, t('protected:birth-details.city.invalid-city')),
+                v.regex(REGEX_PATTERNS.NON_DIGIT, t('protected:birth-details.city.invalid-city')),
+              ),
+            ),
+            fromMultiple: v.boolean(t('protected:birth-details.from-multiple.required-from-multiple')),
+          }),
+        ],
+        t('protected:birth-details.country.required-country'),
+      ) satisfies v.GenericSchema<BirthDetailsSessionData>;
+
+      const input = {
+        country: formData.get('country') as string,
+        province: trimToUndefined(formData.get('province') as string),
+        city: trimToUndefined(formData.get('city') as string),
+        fromMultiple: formData.get('from-multiple') ? formData.get('from-multiple') === REQUIRE_OPTIONS.yes : undefined,
+      } satisfies Partial<v.InferInput<typeof schema>>;
+
+      const parseResult = v.safeParse(schema, input, { lang });
+
+      if (!parseResult.success) {
+        return data({ errors: v.flatten<typeof schema>(parseResult.issues).nested }, { status: 400 });
+      }
+
+      context.session.inPersonSINCase ??= {};
+      context.session.inPersonSINCase.birthDetails = parseResult.output;
+
+      throw i18nRedirect('routes/protected/request.tsx', request);
+    }
+
+    default: {
+      throw new AppError(`Unrecognized action: ${action}`, ErrorCodes.UNRECOGNIZED_ACTION);
+    }
+  }
+}
+
+export default function BirthDetails({ loaderData, actionData, params }: Route.ComponentProps) {
+  const { t } = useTranslation(handle.i18nNamespace);
+  const [country, setCountry] = useState<string | undefined>(loaderData.defaultFormValues.firstName);
+  const [fromMultiple, setFromMultiple] = useState<boolean | undefined>(
+    loaderData.defaultFormValues.supportingDocumentsRequired,
+  );
+  const fetcherKey = useId();
+  const fetcher = useFetcher<Info['actionData']>({ key: fetcherKey });
+  const isSubmitting = fetcher.state !== 'idle';
+  const errors = fetcher.data?.errors;
+  const canadaCountryCode = 'CAN';
+
+  function handleCountryChange(event: ChangeEvent<HTMLSelectElement>) {
+    setCountry(event.target.value);
+  }
+
+  const countryOptions = ['select-option', ...COUNTRIES].map((value) => ({
+    value: value === 'select-option' ? '' : value,
+    children: t(`protected:birth-details.countries.${value}` as 'protected:birth-details.countries.select-option'),
+  }));
+
+  function handleFromMultipleChanged(event: ChangeEvent<HTMLInputElement>) {
+    setFromMultiple(event.target.value === REQUIRE_OPTIONS.yes);
+  }
+
+  const fromMultipleOptions: InputRadiosProps['options'] = [
+    {
+      children: t('gcweb:input-option.yes'),
+      value: REQUIRE_OPTIONS.yes,
+      defaultChecked: fromMultiple === true,
+      onChange: handleFromMultipleChanged,
```

Comment:
You could save yourself a function definition and remove some clutter if you do this:

``` tsx
onChange: ({ target }) => setFromMultiple(target.value === REQUIRE_OPTIONS.yes),
```